### PR TITLE
LG-3813: Alternate row color overrides selected row color

### PR DESCRIPTION
**Pull request by spark33**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3813](https://jira.mongodb.org/browse/LG-3813)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/table/src/Row/InternalRowWithRT.tsx [hunk 1]
```diff
@@ -87,7 +87,8 @@ const InternalRowWithRT = <T extends LGRowData>({
         className={cx(
           {
             [grayZebraRowStyles[theme]]: isOddVSRow && shouldAlternateRowColor,
-            [zebraStyles[theme]]: !virtualRow && shouldAlternateRowColor,
+            [zebraStyles[theme]]:
+              !virtualRow && shouldAlternateRowColor && !isSelected,
```
Comment (TheSonOfThomp) [inline packages/table/src/Row/InternalRowWithRT.tsx:91]: @spark33 Should we be applying this condition to the above styles too? (`grayZebraRowStyles`)

File: packages/table/src/Row/InternalRowWithRT.tsx [hunk 2]
```diff
@@ -86,8 +86,10 @@ const InternalRowWithRT = <T extends LGRowData>({
       <InternalRowBase
         className={cx(
           {
-            [grayZebraRowStyles[theme]]: isOddVSRow && shouldAlternateRowColor,
-            [zebraStyles[theme]]: !virtualRow && shouldAlternateRowColor,
+            [grayZebraRowStyles[theme]]:
+              isOddVSRow && shouldAlternateRowColor && !isSelected,
```
Comment (TheSonOfThomp) [inline packages/table/src/Row/InternalRowWithRT.tsx:90]: nit: should we break `shouldAlternateRowColor && !isSelected` into a separate variable?

Comment (TheSonOfThomp) [reply packages/table/src/Row/InternalRowWithRT.tsx:90]: Also, does this style depend on `!virtualRow` as well?

**Review by TheSonOfThomp — APPROVED** [review]
Left a few nits, but they're non-blocking

File: packages/table/src/Row/InternalRowWithRT.tsx (hunk 2 above)
Comment (spark33) [reply packages/table/src/Row/InternalRowWithRT.tsx:90]: `isOddVSRow` should imply `!!virtualRow`, so shouldn't be necessary!

**Pull request merged**